TestFoundation/TestNSByteCountFormatter.swift

```diff
@@ -39,4 +45,131 @@ class TestNSByteCountFormatter : XCTestCase {
+        
+        formatter.allowedUnits = .useDefault
+        formatter.includesCount = false
+        XCTAssertEqual(formatter.string(fromByteCount: 100), "bytes")
+        XCTAssertEqual(formatter.string(fromByteCount: 1300), "KB")
+        
+        
+    }
+    
+    func test_isAdaptive() {
+        let formatter = ByteCountFormatter()
+        formatter.isAdaptive = false
+        
+        XCTAssertEqual(formatter.string(fromByteCount: 123000), "123 KB")
+        
+        //isAdaptive tests for when allowUnits is set
+        formatter.allowedUnits = .useTB
+        XCTAssertEqual(formatter.string(fromByteCount: 10000000), "0.0000100 TB")
+    }
+    
+    func test_zeroPadsFractionDigits() {
+        let formatter = ByteCountFormatter()
+        formatter.zeroPadsFractionDigits = true
+        
+        XCTAssertEqual(formatter.string(fromByteCount: 120), "120 bytes")
+        XCTAssertEqual(formatter.string(fromByteCount: 12000), "12 KB")
+        XCTAssertEqual(formatter.string(fromByteCount: 12000000), "12.0 MB")
+        XCTAssertEqual(formatter.string(fromByteCount: 12000000000), "12.00 GB")
+        
+        formatter.isAdaptive = false
+        XCTAssertEqual(formatter.string(fromByteCount: 1200), "1.20 KB")
+        XCTAssertEqual(formatter.string(fromByteCount: 12000), "12.0 KB")
+        XCTAssertEqual(formatter.string(fromByteCount: 12000000), "12.0 MB")
+        XCTAssertEqual(formatter.string(fromByteCount: 12000000000), "12.0 GB")
+    }
+    
+    func test_allowedUnits() {
+        let formatter = ByteCountFormatter()
+        formatter.allowedUnits = .useBytes
+        XCTAssertEqual(formatter.string(fromByteCount: 12000000000000), "12,000,000,000,000 bytes")
+        
+        formatter.allowedUnits = .useKB
+        XCTAssertEqual(formatter.string(fromByteCount: 12000000000000), "12,000,000,000 KB")
+        
+        formatter.allowedUnits = .useMB
+        XCTAssertEqual(formatter.string(fromByteCount: 12000000000000), "12,000,000 MB")
+        
+        formatter.allowedUnits = .useGB
+        XCTAssertEqual(formatter.string(fromByteCount: 12000000000000), "12,000 GB")
+        
+        formatter.allowedUnits = .useTB
+        XCTAssertEqual(formatter.string(fromByteCount: 12000000000000), "12 TB")
+        
+        formatter.allowedUnits = .usePB
+        XCTAssertEqual(formatter.string(fromByteCount: 12000000000000), "0.01 PB")
+        
+        formatter.allowedUnits = .useEB
+        XCTAssertEqual(formatter.string(fromByteCount: 1900000000000000000), "1.9 EB")
+        
+        formatter.allowedUnits = .useZB
+        XCTAssertEqual(formatter.string(fromByteCount: 6000000000000000000), "0.01 ZB")
+        
+        formatter.allowedUnits = .useYBOrHigher
+        XCTAssertEqual(formatter.string(fromByteCount: 6000000000000000000), "0 YB")
+        XCTAssertEqual(formatter.string(fromByteCount: 0), "Zero KB")
+    }
+
```

Comment:
Perfect, thanks for supplying that. 
At a quick glance I noticed a few cases the initial implementation is missing. Particularly involving the OptionSet. I've begun creating the tests and fixing the issues raised by them, I'll update once I've finished. 

Thanks again.